panther_manager/src/manager_bt_node.cpp

```diff
@@ -90,58 +92,73 @@ ManagerBTNode::ManagerBTNode(
-    {"CRITICAL_BATTERY_THRESHOLD_PERCENT", critical_battery_threshold_percent},
-    {"BATTERY_STATE_ANIM_PERIOD", battery_state_anim_period},
-    {"LOW_BATTERY_ANIM_PERIOD", low_battery_anim_period},
-    {"LOW_BATTERY_THRESHOLD_PERCENT", low_battery_threshold_percent},
-    {"current_anim_id", -1},
-    {"charging_anim_percent", ""},
-    // anim constants
-    {"E_STOP_ANIM_ID", unsigned(panther_msgs::LEDAnimation::E_STOP)},
-    {"READY_ANIM_ID", unsigned(panther_msgs::LEDAnimation::READY)},
-    {"ERROR_ANIM_ID", unsigned(panther_msgs::LEDAnimation::ERROR)},
-    {"MANUAL_ACTION_ANIM_ID", unsigned(panther_msgs::LEDAnimation::MANUAL_ACTION)},
-    {"AUTONOMOUS_ACTION_ANIM_ID", unsigned(panther_msgs::LEDAnimation::AUTONOMOUS_ACTION)},
-    {"GOAL_ACHIEVED_ANIM_ID", unsigned(panther_msgs::LEDAnimation::GOAL_ACHIEVED)},
-    {"LOW_BATTERY_ANIM_ID", unsigned(panther_msgs::LEDAnimation::LOW_BATTERY)},
-    {"CRITICAL_BATTERY_ANIM_ID", unsigned(panther_msgs::LEDAnimation::CRITICAL_BATTERY)},
-    {"BATTERY_STATE_ANIM_ID", unsigned(panther_msgs::LEDAnimation::BATTERY_STATE)},
-    {"CHARGING_BATTERY_ANIM_ID", unsigned(panther_msgs::LEDAnimation::CHARGING_BATTERY)},
-    // battery constants
-    {"POWER_SUPPLY_STATUS_UNKNOWN",
-     unsigned(sensor_msgs::BatteryState::POWER_SUPPLY_STATUS_UNKNOWN)},
-    {"POWER_SUPPLY_STATUS_CHARGING",
-     unsigned(sensor_msgs::BatteryState::POWER_SUPPLY_STATUS_CHARGING)},
-    {"POWER_SUPPLY_STATUS_DISCHARGING",
-     unsigned(sensor_msgs::BatteryState::POWER_SUPPLY_STATUS_DISCHARGING)},
-    {"POWER_SUPPLY_STATUS_NOT_CHARGING",
-     unsigned(sensor_msgs::BatteryState::POWER_SUPPLY_STATUS_NOT_CHARGING)},
-    {"POWER_SUPPLY_STATUS_FULL", unsigned(sensor_msgs::BatteryState::POWER_SUPPLY_STATUS_FULL)},
-  };
-  const std::map<std::string, std::any> safety_initial_bb = {
-    {"HIGH_BAT_TEMP", high_bat_temp},
-    {"CRITICAL_BAT_TEMP", critical_bat_temp},
-    {"FATAL_BAT_TEMP", fatal_bat_temp},
-    {"CPU_FAN_ON_TEMP", cpu_fan_on_temp},
-    {"CPU_FAN_OFF_TEMP", cpu_fan_off_temp},
-    {"DRIVER_FAN_ON_TEMP", driver_fan_on_temp},
-    {"DRIVER_FAN_OFF_TEMP", driver_fan_off_temp},
-  };
-  const std::map<std::string, std::any> shutdown_initial_bb = {
-    {"SHUTDOWN_HOSTS_FILE", shutdown_hosts_file.c_str()},
-  };
-
-  lights_config_ = create_bt_config(lights_initial_bb);
-  safety_config_ = create_bt_config(safety_initial_bb);
-  shutdown_config_ = create_bt_config(shutdown_initial_bb);
-
-  lights_tree_ = factory_.createTree("Lights", lights_config_.blackboard);
-  safety_tree_ = factory_.createTree("Safety", safety_config_.blackboard);
-  shutdown_tree_ = factory_.createTree("Shutdown", shutdown_config_.blackboard);
+  factory_.registerBehaviorTreeFromFile(bt_project_file);
+
+  if (launch_safety_tree && !launch_shutdown_tree_) {
+    ROS_ERROR(
+      "[%s] Can't launch safety tree without shutdown tree. Killing node.", node_name_.c_str());
+    ros::requestShutdown();
+  }
```

Comment:
I don't know how BT looks like, so I don't understand why this should be met. I see there is empty space in README.md for [BT}(https://github.com/husarion/panther_ros/tree/ros1-manager-bt-update/panther_manager#behaviortree)

Reply:
One of Safety tree tasks is to signal robot shutdown in case of reaching fatal battery temperature.  This requires the Shutdown tree to is present.